### PR TITLE
feat(fab): typescript support

### PR DESCRIPTION
**Pull request by moog16**
related #48

### CODE REVIEW

File: packages/fab/index.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    return null;
+  }
+  const updatedProps = {
+    className: classnames('mdc-fab__icon', icon.props.className),
+  };
+  return React.cloneElement(icon, updatedProps);
+};
+
+const TextLabel: React.FunctionComponent<{textLabel: string}> = ({
+  textLabel, // eslint-disable-line react/prop-types
```
Comment (williamernest) [inline packages/fab/index.tsx]: Are the eslint comments a temporary measure until we completely convert everything? 

We may want to make another issue to install [google/ts-tyle](https://github.com/google/ts-style) into this repo.

File: packages/fab/index.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+import withRipple from '@material/react-ripple';
+
+export interface FabProps {
+  mini: boolean,
+  icon?: React.ReactElement<HTMLElement>,
+  textLabel: string,
+  className: string,
+  initRipple: (surface: HTMLButtonElement) => void,
+  unbounded: boolean,
+};
```
Comment (williamernest) [inline packages/fab/index.tsx]: Nit: semicolon isn't needed at the end of interfaces.

File: packages/fab/index.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+import * as React from 'react';
```
Comment (williamernest) [inline packages/fab/index.tsx]: Nit: Add new line after license.

File: test/screenshot/fab/index.tsx [hunk 4]
```diff
@@ -0,0 +1,250 @@
+import * as React from 'react';
+// @ts-ignore
```
Comment (williamernest) [inline test/screenshot/fab/index.tsx]: I'm not sure why the Fab import has to specify .tsx (which is the error being suppressed here). Can you document why this is required in a comment?

File: packages/fab/index.tsx (hunk 1 above)
Comment (moog16) [reply packages/fab/index.tsx]: ya temporary. I didn't want to break all the other js files until all are converted to TS

Comment (moog16) [reply packages/fab/index.tsx]: https://github.com/material-components/material-components-web-react/issues/507

File: test/screenshot/fab/index.tsx (hunk 4 above)
Comment (moog16) [reply test/screenshot/fab/index.tsx]: This is because we still have the index.js files. This is temporary until we get all TSX files in.

**Pull request merged**